tests/test_run_modules.py

```diff
@@ -99,13 +99,13 @@ def test_run_topostats_process_all(caplog) -> None:
     # pytest was invoked with (see thread on StackOverflow at https://stackoverflow.com/a/55260580/1444043)
     entry_point(
         manually_provided_args=[
-            "process",
             "--config",
             f"{BASE_DIR / 'topostats' / 'default_config.yaml'}",
             "--base-dir",
             "./tests/resources/test_image/",
             "--file-ext",
             ".topostats",
+            "process",
```

Comment:
Its because these common arguments are for the parser rather than the various `subparsers` so I thought it made sense to have them before the `program` argument...

```
❱ topostats --help
usage: topostats [-h] [-v] [-c CONFIG_FILE] [-s SUMMARY_CONFIG] [--matplotlibrc MATPLOTLIBRC] [-b BASE_DIR] [-o OUTPUT_DIR] [-l LOG_LEVEL] [-j CORES] [-f FILE_EXT] [--channel CHANNEL]
                 {process,load,filter,grains,grainstats,disordered-tracing,nodestats,ordered-tracing,splining,summary,create-config,create-matplotlibrc} ...

Run various programs relating to AFM data. Add the name of the program you wish to run.

options:
  -h, --help            show this help message and exit
  -v, --version         Report the current version of TopoStats that is installed
  -c CONFIG_FILE, --config-file CONFIG_FILE
                        Path to a YAML configuration file.
  -s SUMMARY_CONFIG, --summary-config SUMMARY_CONFIG
                        Path to a YAML configuration file for summary plots and statistics.
  --matplotlibrc MATPLOTLIBRC
                        Path to a matplotlibrc file.
  -b BASE_DIR, --base-dir BASE_DIR
                        Base directory to scan for images.
  -o OUTPUT_DIR, --output-dir OUTPUT_DIR
                        Output directory to write results to.
  -l LOG_LEVEL, --log-level LOG_LEVEL
                        Logging level to use, default is 'info' for verbose output use 'debug'.
  -j CORES, --cores CORES
                        Number of CPU cores to use when processing.
  -f FILE_EXT, --file-ext FILE_EXT
                        File extension to scan for.
  --channel CHANNEL     Channel to extract.

program:
  Available programs, listed below:

  {process,load,filter,grains,grainstats,disordered-tracing,nodestats,ordered-tracing,splining,summary,create-config,create-matplotlibrc}
    process             Process AFM images. Additional arguments over-ride defaults or those in the configuration file.
    load                Load and save all images as .topostats files for subsequent processing.
    filter              WIP DO NOT USE - Load and filter images, saving as .topostats files for subsequent processing.
    grains              WIP DO NOT USE - Load filtered images from '.topostats' files and detect grains.
    grainstats          WIP DO NOT USE - Load images with grains from '.topostats' files and calculate statistics.
    disordered-tracing  WIP DO NOT USE - Skeletonise and prune objects to disordered traces.
    nodestats           WIP DO NOT USE - Calculate node statistics and disentangle molecules.
    ordered-tracing     WIP DO NOT USE - Ordered traces of pruned skeletons.
    splining            WIP DO NOT USE - Splining of traced molecules to produce smooth curves.
    summary             Plotting and summary of TopoStats output statistics.
    create-config       Create a configuration file using the defaults.
    create-matplotlibrc
                        Create a Matplotlibrc parameters file using the defaults.
```

You then access the arguments specific to the program specifying them after the program.

```
 ❱ topostats filter --help
usage: topostats filter [-h] [--row-alignment-quantile ROW_ALIGNMENT_QUANTILE] [--threshold-method THRESHOLD_METHOD] [--otsu-threshold-multiplier OTSU_THRESHOLD_MULTIPLIER] [--threshold-std-dev-below THRESHOLD_STD_DEV_BELOW] [--threshold-std-dev-above THRESHOLD_STD_DEV_ABOVE]
                        [--threshold-absolute-below THRESHOLD_ABSOLUTE_BELOW] [--threshold-absolute-above THRESHOLD_ABSOLUTE_ABOVE] [--gaussian-size GAUSSIAN_SIZE] [--gaussian-mode GAUSSIAN_MODE] [--remove-scars SCARS_RUN] [--scars-removal-iterations SCARS_REMOVAL_ITERATIONS]
                        [--scars-threshold-low SCARS_THRESHOLD_LOW] [--scars-threshold-high SCARS_THRESHOLD_HIGH] [--scars-max-scar-width SCARS_MAX_SCAR_WIDTH] [--scars-max-scar-length SCARS_MAX_SCAR_LENGTH]

WIP DO NOT USE - Load and filter images, saving as .topostats files for subsequent processing.

options:
  -h, --help            show this help message and exit
  --row-alignment-quantile ROW_ALIGNMENT_QUANTILE
                        Lower values may improve flattening of larger features.
  --threshold-method THRESHOLD_METHOD
                        Method for thresholding Filtering. Options are otsu, std_dev, absolute.
  --otsu-threshold-multiplier OTSU_THRESHOLD_MULTIPLIER
                        Factor for scaling the Otsu threshold during Filtering.
  --threshold-std-dev-below THRESHOLD_STD_DEV_BELOW
                        Threshold for data below the image background for std dev method during Filtering.
  --threshold-std-dev-above THRESHOLD_STD_DEV_ABOVE
                        Threshold for data above the image background for std dev method during Filtering.
  --threshold-absolute-below THRESHOLD_ABSOLUTE_BELOW
                        Threshold for data below the image bacnground dor absolute method during Filtering
  --threshold-absolute-above THRESHOLD_ABSOLUTE_ABOVE
                        Threshold for data above the image bacnground dor absolute method during Filtering
  --gaussian-size GAUSSIAN_SIZE
                        Gaussian blur intensity in pixels.
  --gaussian-mode GAUSSIAN_MODE
                        Gaussian blur method. Options are 'nearest' (default), 'reflect', 'constant', 'mirror' or 'wrap'.
  --remove-scars SCARS_RUN
                        Whether to remove scars.
  --scars-removal-iterations SCARS_REMOVAL_ITERATIONS
                        Number of times to run scar removal
  --scars-threshold-low SCARS_THRESHOLD_LOW
                        Lower values make scar removal more sensitive
  --scars-threshold-high SCARS_THRESHOLD_HIGH
                        Lower values make scar removal more sensitive
  --scars-max-scar-width SCARS_MAX_SCAR_WIDTH
                        Maximum thickness of scars in pixels
  --scars-max-scar-length SCARS_MAX_SCAR_LENGTH
                        Maximum length of scars in pixels
```

Reply:
Makes sense! 👍